src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -49,67 +48,144 @@ class DetectorStateData:
     counter_updates: dict[str, int | None]
 
 
+# TODO - refactor the `DetectorGroup` information out of this layer.
```

Comment:
This is something i think we could do or could skip -- in pt3 i'm building a stateful handler using this state class w/ `None` as the group key, and that would also encapsulate the grouping logic so implementing handlers don't worry about it.